### PR TITLE
RD-5205 part 2: make headers consistent in  agents widget

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
part 2 of RD-5205. the order of the PRs doesn't matter.

## Screenshots / Videos
Screen Shot 2022-07-14 at 10 11 26

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests
Latest:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3128/pipeline

I started `agents_spec` in this branch in Jenkins: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3123/pipeline

initial, failed because of Jenkins issues:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3117/pipeline

 

## Documentation
https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2247

### CODE REVIEW

File: widgets/agents/src/AgentsTable.tsx [hunk 1]
```diff
@@ -3,14 +3,35 @@
 import InstallAgentsModal from './InstallAgentsModal';
 import ValidateAgentsModal from './ValidateAgentsModal';
 import AgentsPropType from './props/AgentsPropType';
+import type { Agents } from './types';
 
-export default class AgentsTable extends React.Component {
+const t = Stage.Utils.getT('widgets.agents');
```
Comment (qooban) [inline widgets/agents/src/AgentsTable.tsx:8]: There's no `widgets.agents` in `en.json`. Please check.

**Review by qooban — COMMENTED** [review]
Added a comment.

Tests - seems like you need to re-run the job as it failed (looks like env issue).
Documentation - are you planning to update it at once?

File: widgets/agents/src/AgentsTable.tsx (hunk 1 above)
Comment (leon-good-life) [reply widgets/agents/src/AgentsTable.tsx:8]: I addressed this review comment, I made 2 commits in which I added the translation: [d41bb92](https://github.com/cloudify-cosmo/cloudify-stage/pull/2062/commits/d41bb924fd21f1c05e105d881a561ebe0b3fb121), [4ad096e](https://github.com/cloudify-cosmo/cloudify-stage/pull/2062/commits/4ad096e850847a8b91feb1ca95aaa66b9859fde4)

I opened the UI and verified: 
Screen Shot 2022-07-14 at 10 11 26

**Comment by leon-good-life** [comment]
I started `agents_spec` in this branch in Jenkins: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3123/pipeline

**Comment by leon-good-life** [comment]
I updated the docs repo: https://github.com/cloudify-cosmo/docs.getcloudify.org/pull/2247

**Comment by leon-good-life** [comment]
I fixed `agents_spec` test, and started it again on Jenkins: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3125/pipeline

**Comment by leon-good-life** [comment]
I fixed the test and started the Jenkins again: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3128/pipeline

**Comment by leon-good-life** [comment]
The test passed

File: widgets/agents/src/widget.tsx [hunk 2]
```diff
@@ -28,8 +28,8 @@ Stage.defineWidget({
                 'Some of the fields may be hidden depending on the context, ' +
                 'eg. when Deployment ID is set in context then Deployment field will be hidden.',
             placeHolder: 'Select fields from the list',
-            items: ['Id', 'Node', 'Deployment', 'IP', 'Install Method', 'System', 'Version', 'Actions'],
-            default: 'Id,Node,Deployment,IP,Install Method,System,Version,Actions',
+            items: ['ID', 'Node', 'Deployment', 'IP', 'Install method', 'System', 'Version', 'Actions'],
+            default: 'ID,Node,Deployment,IP,Install method,System,Version,Actions',
```
Comment (qooban) [inline widgets/agents/src/widget.tsx:32]: Can we get it from translation file? We already do that in Blueprints widget or Blueprints Catalog widget. It should make it less prone to errors.

File: widgets/agents/src/AgentsTable.tsx [hunk 3]
```diff
@@ -3,14 +3,35 @@
...
+        deploymentId: string | string[];
+        nodeId: string | string[];
+        nodeInstanceId: string | string[];
+    };
+    widget: Stage.Types.Widget;
+    toolbox: Stage.Types.Toolbox;
+}
+
+interface AgentsTableState {
+    error: any;
```
Comment (qooban) [inline widgets/agents/src/AgentsTable.tsx:23]: Can we be more precise with the type here?

Comment (qooban) [reply widgets/agents/src/AgentsTable.tsx:23]: OK, in that case let's remove it from the state and update all relevant places.

**Review by qooban — COMMENTED** [review]
Let's try to resolve outstanding discussion - https://github.com/cloudify-cosmo/cloudify-stage/pull/2062/files#r920963712.

File: widgets/agents/src/AgentsTable.tsx (hunk 3 above)
Comment (leon-good-life) [reply widgets/agents/src/AgentsTable.tsx]: [e8a9b92](https://github.com/cloudify-cosmo/cloudify-stage/pull/2062/commits/e8a9b92f27277187ff0cd8fb1fe0faa957c76f08)

**Comment by leon-good-life** [comment]
I started `agents_spec` on Jenkins: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3129/pipeline

**Comment by leon-good-life** [comment]
The system test passed

**Review by qooban — APPROVED** [review]

**Pull request merged**